### PR TITLE
Add some tests for delete_range

### PR DESCRIPTION
**Pull request by follitude**
PTAL @zhangjinpeng1987

### CODE REVIEW

File: tests/test_delete_range.rs [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+        let a = db.get(b"a");
+        assert_eq!(a.unwrap().unwrap(), b"v1");
+        db.put(b"b", b"v2").unwrap();
+        let b = db.get(b"b");
+        assert_eq!(b.unwrap().unwrap(), b"v2");
+        db.put(b"c", b"v3").unwrap();
+        let c = db.get(b"c");
+        assert_eq!(c.unwrap().unwrap(), b"v3");
+    };
+    prepare_data();
```
Comment (siddontang) [inline tests/test_delete_range.rs]: why use a local closure function here?

Comment (follitude) [inline tests/test_delete_range.rs]: This is a previous test written by someone, I just keep it.

Comment (siddontang) [inline tests/test_delete_range.rs]: got it

File: tests/test_delete_range.rs [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    db.flush(true).unwrap();
+
+    db.compact_range(None, None);
+    assert_eq!(db.get(b"key3").unwrap().unwrap(), b"value5");
+
+    db.delete_range(b"key1", b"key1").unwrap();
+    assert_eq!(db.get(b"key1").unwrap().unwrap(), b"value1");
+
+    db.delete_range(b"key2", b"key7").unwrap();
+    assert!(db.get(b"key2").unwrap().is_none());
```
Comment (siddontang) [inline tests/test_delete_range.rs]: you can use table-driven check here.

File: tests/test_delete_range.rs [hunk 3]
```diff
@@ -0,0 +1,185 @@
+// Copyright 2014 Tyler Neely
```
Comment (zhangjinpeng1987) [inline tests/test_delete_range.rs]: Please update the license

File: tests/test_delete_range.rs [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+                         (b"key2".to_vec(), b"value2".to_vec()),
+                         (b"key3".to_vec(), b"value3".to_vec()),
+                         (b"key4".to_vec(), b"value4".to_vec())];
+    for &(ref k, ref v) in &samples_a {
+        db.put(k, v).unwrap();
+        assert_eq!(v.as_slice(), &*db.get(k).unwrap().unwrap());
+    }
+    db.flush(true).unwrap();
+
+    let samples_b = vec![(b"key3".to_vec(), b"value5".to_vec()),
```
Comment (zhangjinpeng1987) [inline tests/test_delete_range.rs]: s/key3/key5

File: tests/test_delete_range.rs [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+                         (b"key6".to_vec(), b"value6".to_vec()),
+                         (b"key7".to_vec(), b"value7".to_vec()),
+                         (b"key8".to_vec(), b"value8".to_vec())];
+    for &(ref k, ref v) in &samples_b {
+        db.put(k, v).unwrap();
+        assert_eq!(v.as_slice(), &*db.get(k).unwrap().unwrap());
+    }
+    db.flush(true).unwrap();
+
+    db.compact_range(None, None);
```
Comment (zhangjinpeng1987) [inline tests/test_delete_range.rs]: What is the purpose to compact range here?

**Comment by zhangjinpeng1987** [comment]
rest LGTM

File: tests/test_delete_range.rs [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+    batch.delete_range_cf(cf_handle, b"a", b"c").unwrap();
+    assert!(db.write(batch).is_ok());
+    check_data();
+}
+
+#[test]
+fn test_delete_range_sst_files() {
+    let path = TempDir::new("_rust_rocksdb_test_delete_range_sst_files").expect("");
+    let db = DB::open_default(path.path().to_str().unwrap()).unwrap();
+    let samples_a = vec![(b"key1".to_vec(), b"value1".to_vec()),
```
Comment (siddontang) [inline tests/test_delete_range.rs]: seem you don't need to_vec here

File: tests/test_delete_range.rs [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+        db.put(k, v).unwrap();
+        assert_eq!(v, &*db.get(k).unwrap().unwrap());
+    }
+    db.flush(true).unwrap();
+
+    let samples_b = vec![(b"key3", b"value5"),
+                         (b"key6", b"value6"),
+                         (b"key7", b"value7"),
+                         (b"key8", b"value8")];
+    for &(k, v) in &samples_b {
```
Comment (siddontang) [inline tests/test_delete_range.rs]: seem you can use `for (k,v ) in samples_b` directly

**Comment by follitude** [comment]
Done. @siddontang

**Comment by siddontang** [comment]
LGTM

**Comment by siddontang** [comment]
PTAL @zhangjinpeng1987

**Comment by zhangjinpeng1987** [comment]
LGTM

**Pull request merged**